cypress/e2e/user-flows/scenario_008.cy.js

```diff
@@ -0,0 +1,108 @@
+import { SimulatorPage } from "../../pages/simulator/simulatorPage";
+import { EditorPage } from "../../pages/simulator/editorPage";
+import { SlicePage } from "../../pages/slices/slicePage";
+import { SLICE_MOCK_FILE } from "../../consts";
+
+const SLICE = {
+  id: "scenario008",
+  name: "Scenario008",
+  library: "slices",
+};
+
+describe("Scenario 008", () => {
+  let slice = new SlicePage();
+  let simulator = new SimulatorPage();
+  let editor = new EditorPage();
+
+  beforeEach(() => {
+    cy.clearProject();
+    cy.setSliceMachineUserContext({});
+  });
+
+  it("creates slice and tests the editor mocks", () => {
+    cy.createSlice(SLICE.library, SLICE.id, SLICE.name);
+
+    slice
+      .addVariation("SecondVariation")
+      .addNewWidgetField("SimpleTextField", "Key Text")
+      .addNewWidgetField("RichTextField", "Rich Text")
+      .addNewWidgetField("NumberField", "Number")
+      .addNewWidgetField("BooleanField", "Boolean")
+      .addNewWidgetField("SelectField", "Select")
+      .addNewWidgetField("ImageField", "Image")
+      .save();
+
+    // force mock value for the boolean and select fields
+    cy.modifyFile(SLICE_MOCK_FILE(SLICE.name), (mock) =>
+      mock.map((variation) =>
+        variation.variation === "default"
+          ? variation
+          : {
+              ...variation,
+              primary: {
+                ...variation.primary,
+                booleanfield: {
+                  ...variation.primary.booleanfield,
+                  value: false,
+                },
+                selectfield: {
+                  ...variation.primary.selectfield,
+                  value: "1",
+                },
+              },
+            }
+      )
+    );
+
+    simulator.setup();
+    slice.openSimulator();
+
+    // Wait for the editor to be fully loaded
+    cy.contains("label > span", "Title").should("be.visible");
+
+    simulator.changeVariations("Default");
+    cy.contains("Scenario008 • Default").should("be.visible");
+
+    simulator.changeVariations("SecondVariation");
+
+    // editor # toggle
+    cy.contains("Scenario008 • SecondVariation").should("be.visible");
+    simulator.toggleEditor();
+    cy.contains("Scenario008 • SecondVariation").should("not.be.visible");
+    simulator.toggleEditor();
+    cy.contains("Scenario008 • SecondVariation").should("be.visible");
+
+    // editor # change field values
+    editor.type("SimpleTextField", "SimpleTextContent");
+    editor.type("RichTextField", "RichTextContent");
+    editor.type("NumberField", "42", "have.value");
+    editor.toggleBooleanField("BooleanField");
+    editor.select("SelectField", "2");
+
+    editor.changeImage("ImageField").then((newImageSrc) => {
```

Comment:
If you want to avoid having to nest the rest of the test just to access `newImageSrc`, you could use an alias instead:
```js
    editor.changeImage("ImageField").invoke('attr', 'src').as('newImageSrc');
```
Then write the rest of the test without nesting it, and using the alias only when you need it:
```js
    cy.get("@newImageSrc").then(newImageSrc => {
      cy.get("[alt='An ananas maybe']")
        .invoke("attr", "src")
        .should("contain", newImageSrc.substr(0, newImageSrc.indexOf("?") + 1));
    })
```
For that to work, you would also have to update the `changeImage()` method to remove the last line `.then(([img]) => img.src)`. It would actually be less specific in case other tests want to access another attribute.

It's just an idea, I'm not sure which approach is more readable.

Reply:
Nice, it worked!

Reply:
👌